### PR TITLE
Add PseudoLabeling transform for unlabeled dataset

### PR DESCRIPTION
**Pull request by sooahleex**
### Summary

- Add `PseudoLabeling` transform for unlabeled dataset
- Use `Explorer` of Datumaro to make pseudo label for each datasetitem

### How to test

- Add unit tests

### Checklist

- [X] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [X] I have added the description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md).​
- [X] I have updated the [documentation](https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly

### License

- [ ] I submit _my code changes_ under the same [MIT License](https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: src/datumaro/plugins/transforms.py [hunk 1]
```diff
@@ -2004,3 +2007,64 @@ def transform_item(self, item):
...
+    def transform_item(self, item: DatasetItem):
+        hashkey_ = np.unpackbits(self._explorer._get_hash_key_from_item_query(item).hash_key)
+        logits = calculate_hamming(hashkey_, self._label_hashkeys)
+        inverse_distances = 1.0 / (logits + 1e-6)
+        probs = inverse_distances / np.sum(inverse_distances)
+        ind = np.argsort(probs)[::-1]
+
+        pseudo = np.array(self._labels)[ind][0]
+        pseudo_annotation = [Label(label=self._label_indices[pseudo])]
+        return self.wrap_item(item, annotations=pseudo_annotation)
```
Comment (jihyeonyi) [inline src/datumaro/plugins/transforms.py:2070]: It seems that the original annotations are replaced with the pseudo annotation. Is it intended? Then how about add this notice to the document?

Comment (sooahleex) [reply src/datumaro/plugins/transforms.py:2070]: In this case, I assumed the case that each item does not have any annotation. so the original annotations is not existed. If my assumption is not proper, please let me know.

Comment (jihyeonyi) [reply src/datumaro/plugins/transforms.py:2070]: Oh, I see. Then, we can enhance this feature when the assumption is changed.

**Review by jihyeonyi — APPROVED** [review]
